File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/BrushOperation/EntitySpawnOperation.java
package com.github.kevindagame.util.BrushOperation;

import com.github.kevindagame.voxelsniper.entity.entitytype.VoxelEntityType;
import com.github.kevindagame.voxelsniper.location.BaseLocation;

/**
 * Operation that is performed on an entity. This operation spawns an entity
 */
public class EntitySpawnOperation extends BrushOperation {

    private VoxelEntityType entityType;

    public EntitySpawnOperation(BaseLocation location, VoxelEntityType entityType) {
        super(location);
        this.entityType = entityType;
    }

    public VoxelEntityType getEntityType() {
        return entityType;
    }

    public void setEntityType(VoxelEntityType entityType) {
        this.entityType = entityType;
    }
}

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/brushOperation/operation/EntitySpawnOperation.kt
package com.github.kevindagame.util.brushOperation.operation

import com.github.kevindagame.snipe.Undo
import com.github.kevindagame.voxelsniper.entity.entitytype.VoxelEntityType
import com.github.kevindagame.voxelsniper.location.BaseLocation

/**
 * Operation that is performed on an entity. This operation spawns an entity
 */
class EntitySpawnOperation(location: BaseLocation, var entityType: VoxelEntityType) : BrushOperation(location) {

    override fun perform(undo: Undo): Boolean {
        location.world.spawn(location, entityType)
        return false
    }
}